File: Implement strStr(naive).py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n = len(haystack)
        sn = len(needle)
        if sn==0:
            return 0
        for i in range(n):  # This exact sanme logic doesn't work in cpp ("mississipi" <- "issip")
            if haystack[i]==needle[0]: # Or maybe I didn't write the code properly
                for j in range(sn):
                    if i>=n:
                        return -1
                    if haystack[i]!=needle[j]:
                        break
                    i+=1
                    if j==sn-1:
                        return i-j-1
        return -1

File: Implement strStr(naive).cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = haystack.size();
        int sn = needle.size();
        if (sn==0)
            return 0;
        int temp;
        for (int i=0; i<n; i++){
            if (haystack[i]==needle[0]){
                temp = i;   // needed to store orig val of i 
                for (int j=0; j<sn; j++){
                    if (i>=n)
                        return -1;
                    if (haystack[i]!=needle[j]){
                        i = temp;
                        break;
                    }
                    i++;
                    if (j==sn-1)
                        return i-j-1;
                }
            }
            
        }
        return -1;
    }
};